addons/web_tour/static/src/js/tip.js

```diff
@@ -316,8 +316,25 @@ var Tip = Widget.extend({
                 of: this.$anchor,
                 collision: "none",
                 using: props => {
-                    this.el.style.setProperty('top', `${props.top}px`, 'important');
-                    this.el.style.setProperty('left', `${props.left}px`, 'important');
+                    const {top} = props;
+                    let {left} = props;
+                    const anchor = this.$anchor[0];
+                    if (['P', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'].includes(anchor.nodeName)) {
+                        const textContainerWidth = anchor.getBoundingClientRect().width;
+                        const textNode = anchor.firstChild;
+                        const range = document.createRange();
+                        range.selectNodeContents(textNode);
+                        const textWidth = range.getBoundingClientRect().width;
+
+                        const alignment = window.getComputedStyle(anchor).getPropertyValue('text-align');
+                        if (alignment === 'left') {
+                            left = left - textContainerWidth / 2 + textWidth / 2;
+                        } else if (alignment === 'right') {
+                            left = left + textContainerWidth / 2 - textWidth / 2;
+                        }
```

Comment:
Feature works for "top" or "bottom" tips but worsens the problem for "left" and "right" tips. Handle left and right tips correctly as well please :+1: